document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/mapping/mappers/ApiToDefendantMapper.java

```diff
@@ -10,15 +12,36 @@
 @RequestScope
 @Mapper(componentModel = "spring")
 public interface ApiToDefendantMapper {
+
+    @BeforeMapping
+    default void beforeMapping(@MappingTarget Defendant defendant, DefendantApi defendantApi) {
+        if(defendantApi.getIsCorporateAppointment()) {
+            defendant.setName(defendantApi.getCompanyOfficerDetailsApi().getCompanyName());
+        } else {
+            StringBuilder name = new StringBuilder();
+            String title = defendantApi.getPersonOfficerDetailsApi().getTitle();
+            String forename = defendantApi.getPersonOfficerDetailsApi().getForename();
+            String middleName = defendantApi.getPersonOfficerDetailsApi().getMiddleName();
+            String surname = defendantApi.getPersonOfficerDetailsApi().getSurname();
+
+            if(title != null && title.trim().length() > 0) {
+                name.append(title).append(" ");
+            }
+            if(forename != null && forename.trim().length() > 0) {
+                name.append(forename).append(" ");
+            }
+            if(middleName != null && middleName.trim().length() > 0){
+                name.append(middleName).append(" ");
+            }
+            if(surname != null && surname.trim().length() > 0) {
+                name.append(surname);
```

Comment:
Since we are checking if the trimmed value is empty, should we also be trimming the appended value?
```suggestion
                name.append(surname.trim());
```

Reply:
Done

Reply:
Can we do the same for the other 3 values, not just the surname?

Reply:
Done

Reply:
@rsachdevch did you push this? It does not seem to be done? 😕 Just to be clear, I meant for title, forename and middle name.

Reply:
Might be easier to use Java 8 to write this in a single line which does all the filtering out null, empty or whitespace only, trims then formats into a single String with whitespace between each word.

```
String name = Stream.of(title, forename, middleName, surname)
    .filter(s -> StringUtils.isNotBlank(s))
    .map(String::trim)
    .collect(Collectors.joining(" "));
```

Reply:
Unit test coverage is still missing. I have agreed with the developers I will raise a new PR with the missing unit tests (I have already written them locally to test my suggestion to use the Java 8 stream) because this PR is already complex with lots of changes and comments.